src/main/java/httpserver/route/Route.java

```diff
@@ -0,0 +1,37 @@
+package httpserver.route;
+
+import httpserver.RequestMethod;
+
+public class Route {
+    private RequestMethod requestMethod;
+    private String path;
+    private MethodHandler methodHandler;
+
+
+    public Route setRequestMethod(RequestMethod requestMethod) {
+        this.requestMethod = requestMethod;
+        return this;
```

Comment:
This `getter` `setter` approach is ok - indeed you'll see it in a lot of OO code. The issue is that a `Route` instance can exist before a `path` or `requestMethod` has been set, which places the emphasis on the caller to take care in how to use it.

If you made `Route` immutable then the you can remove a whole class of errors.

e.g.

```
class Route {
    private RequestMethod requestMethod;
    private String path;
    private MethodHandler methodHandler;

   public Route(RequestMethod requestMethod, String path, MethodHandler methodHandler) {
     this.requestMethod = requestMethod;
     this.path = path;
     this.methodHandler = methodHandler;
   }

  // getters
}
```

If you want to build it, then you can have a separate `RouteBuilder` class:

```
class RouteBuilder {
    private RequestMethod requestMethod;
    private String path;
    private MethodHandler methodHandler;

    public RequestBuilder() {
    }

    RouteBuilder requestMethod(RequestMethod requestMethod) {
      this.requestMethod = requestMethod;
      return this;
    }

    RouteBuilder path(String path) {
      this.path = path;
      return this;
   }

  RouteBuilder methodHandler(MethodHandler methodHandler) {
    this.methodHandler = methodHandler;
    return this;
  }

  Route build() {
     // Checks to make sure all values are set
     return new Route(requestMethod, path, methodHandler);
  }
}